modules/lib/lib-portal/src/main/resources/lib/xp/portal.ts

```diff
@@ -27,10 +27,15 @@ export interface SiteConfig<Config> {
     config: Config;
 }
 
+export type Without<T, U> = { [P in Exclude<keyof T, keyof U>]?: never };
+export type XOR<T, U> = T | U extends object ? (Without<T, U> & U) | (Without<U, T> & T) : T | U;
+
+export type IdXorPath = XOR<{id: string}, {path: string}>;
```

Comment:
Correct me if I'm wrong, but maybe this could be expressed in a slightly simpler way, without the `XOR` abstraction:

```suggestion
export type IdXorPath = 
  | {
    id: string;
    path?: never;
  }
  | {
    id?: never;
    path: string;
  };
```

Reply:
That's what I actually did. Unfortunately, IDE/Editor shows the `path` as `undefined` in the suggestions box, so I though that in the end this more complex notation will be better for the developers using types.

Reply:
Ok, I will leave this choice up to you. So I will approve this PR.